src/utils/preprocessing/preprocessing.js

```diff
@@ -1,5 +1,6 @@
+// doesn't remove punctuation when it is part of a word, e.g. "l'Italie" or "it's"
 function removePunctuation(string) {
-    let regex = /[¡!"”“„#$%&'(–)*+,…./—:;«<=>»¿?@[\]^_`{|}~]/g;
+    let regex = /(\s|^)[¡!"”“„#$%&'(–)*+,…./—:;«<=>»¿?@[\]^_`{|}~]+|[¡!"”“„#$%&'(–)*+,…./—:;«<=>»¿?@[\]^_`{|}~]+(\s|$)/g;
```

Comment:
this is something! ^^ :)

Reply:
I would say that we might want to seperate the regex into various ones where each one does one task. 
E.g.:
```
let removeNonLinguisticPunct
let removeTrailSpaces ...
```